packages/plugin-ext/src/main/browser/workspace-main.ts

```diff
@@ -318,6 +327,58 @@ export class WorkspaceMainImpl implements WorkspaceMain, Disposable {
         const canonicalUri = await this.canonicalUriService.provideCanonicalUri(new URI(uri), targetScheme, token);
         return isUndefined(canonicalUri) ? undefined : canonicalUri.toString();
     }
+
+    async $resolveDecoding(resource: UriComponents | undefined, options?: { encoding?: string }): Promise<{ preferredEncoding: string; guessEncoding: boolean }> {
+        const preferredEncoding = await this.getPreferredReadEncoding(resource, options);
+
+        return {
+            preferredEncoding,
+            guessEncoding: this.preferenceService.get('files.autoGuessEncoding', false, resource ? resource.toString() : undefined)
+        };
+    }
+
+    async $resolveEncoding(resource: UriComponents | undefined, options?: { encoding?: string }): Promise<{ encoding: string; hasBOM: boolean }> {
+        let encoding: string;
+        if (resource) {
+            encoding = await this.encodingRegistry.getEncodingForResource(URI.fromComponents(resource), options?.encoding);
+        } else {
+            encoding = options?.encoding ?? UTF8;
+        }
+        // see https://github.com/microsoft/vscode/blob/118f9ecd71a8f101b71ae19e3bf44802aa173209/src/vs/workbench/services/textfile/browser/textFileService.ts#L806
+        const hasBOM = encoding === UTF16be || encoding === UTF16le || encoding === UTF8_with_bom;
+
+        return { encoding, hasBOM };
+    }
+
+    async $validateDetectedEncoding(uri: UriComponents | undefined, detectedEncoding: string | undefined, options: { encoding: string } | undefined): Promise<string> {
```

Comment:
When I read `validate...`, I expect a method that returns an error if the parameters are not valid. Why not `$getValidEncoding`?